qgsquick/from_qgis/qgsquickfeatureslistmodel.cpp

```diff
@@ -21,6 +21,7 @@ QgsQuickFeaturesListModel::QgsQuickFeaturesListModel( QObject *parent )
   : QAbstractListModel( parent ),
     mCurrentLayer( nullptr )
 {
+  QObject::connect( QgsProject::instance(), &QgsProject::cleared, this, &QgsQuickFeaturesListModel::emptyData );
```

Comment:
Cool. It would be worth adding a comment stating this is to avoid dangling pointers of mCurrentLayer and mCurrentFeature (it may not be obvious)